sktime/forecasting/darts.py

```diff
@@ -664,4 +665,182 @@ def get_test_params(cls, parameter_set="default"):
+        The width of the output layer in the future covariate projection residual block.
+        If 0, will bypass feature projection and use the raw feature data.
+    temporal_decoder_hidden : int, optional (default=32)
+        The width of the layers in the temporal decoder.
+    use_layer_norm : bool, optional (default=False)
+        Whether to use layer normalization
+    dropout : float, optional (default=0.1)
+        Dropout probability to be used in fully connected layers.
+    use_static_covariates : bool, optional (default=True)
+        Whether to use static covariates
+    **kwargs
+        Optional arguments to initialize the pytorch_lightning.Module,
+        pytorch_lightning.Trainer.
+
+    References
+    ----------
+    .. [1] https://unit8co.github.io/darts/generated_api/darts.models.forecasting.tide_model.html
+
+    Notes
+    -----
+    If unspecified, all columns will be assumed to be known during predictions duration.
+    """
+
+    _tags = {
+        "authors": ["PranavBhatP"],
+        "maintainers": ["PranavBhatP"],
+        "python_version": ">=3.9",
+        "python_dependencies": ["darts>=0.29"],
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": True,
+        "capability: insample": True,
+        "capability:pred_int": True,
+    }
+
+    def __init__(
+        self,
+        input_chunk_length: int,
+        output_chunk_length: int,
+        output_chunk_shift: int = 0,
+        num_encoder_layers: int = 1,
+        num_decoder_layers: int = 1,
+        decoder_output_dim: int = 16,
+        hidden_size: int = 128,
+        temporal_width_past: int = 4,
+        temporal_width_future: int = 4,
+        temporal_decoder_hidden: int = 32,
+        use_layer_norm: bool = False,
+        dropout: float = 0.1,
+        use_static_covariates: bool = True,
+        **kwargs,
+    ):
+        self.input_chunk_length = input_chunk_length
+        self.output_chunk_length = output_chunk_length
+        self.output_chunk_shift = output_chunk_shift
+        self.use_static_covariates = use_static_covariates
+        self.num_encoder_layers = num_encoder_layers
+        self.num_decoder_layers = num_decoder_layers
+        self.decoder_output_dim = decoder_output_dim
+        self.hidden_size = hidden_size
+        self.temporal_width_past = temporal_width_past
+        self.temporal_width_future = temporal_width_future
+        self.temporal_decoder_hidden = temporal_decoder_hidden
+        self.use_layer_norm = use_layer_norm
+        self.dropout = dropout
+        self.kwargs = kwargs
+
+        super().__init__(
+            input_chunk_length=input_chunk_length,
+            output_chunk_length=output_chunk_length,
+            output_chunk_shift=output_chunk_shift,
+            use_static_covariates=use_static_covariates,
+        )
+
+    def _create_forecaster(self="DartsTiDEModel"):
```

Comment:
```suggestion
    def _create_forecaster(self: "DartsTiDEModel"):
```

Shouldn't it be this? Not reviewing rest of the codebase now due to bandwidth issues, but what does "=" mean here?

Reply:
Yes, thanks for pointing that out, seems to be a minor typo. Will resolve in the following commit.

Reply:
Using `self='DartsTiDEModel` is indeed invalid syntax. I am not yet clear as to why the linters haven't caught it though.